src/input_xml.F90

```diff
@@ -1563,6 +1592,47 @@ subroutine read_geometry_xml()
       call fatal_error("No boundary conditions were applied to any surfaces!")
     end if
 
+    ! Determine opposite side for periodic boundaries
+    do i = 1, size(surfaces)
+      if (surfaces(i) % obj % bc == BC_PERIODIC) then
+        select type (surf => surfaces(i) % obj)
+        type is (SurfaceXPlane)
+          if (i == i_xmin) then
+            surf % opposite = i_xmax
+          elseif (i == i_xmax) then
+            surf % opposite = i_xmin
+          else
+            call fatal_error("Periodic boundary condition applied to &
+                 &interior surface.")
+          end if
+
+        type is (SurfaceYPlane)
+          if (i == i_ymin) then
+            surf % opposite = i_ymax
+          elseif (i == i_ymax) then
+            surf % opposite = i_ymin
+          else
+            call fatal_error("Periodic boundary condition applied to &
+                 &interior surface.")
+          end if
+
+        type is (SurfaceZPlane)
+          if (i == i_zmin) then
+            surf % opposite = i_zmax
+          elseif (i == i_zmax) then
+            surf % opposite = i_zmin
+          else
+            call fatal_error("Periodic boundary condition applied to &
+                 &interior surface.")
+          end if
+
+        class default
+          call fatal_error("Periodic boundary condition applied to &
+               &non-planar surface.")
+        end select
+      end if
+    end do
```

Comment:
Do you think there should also be a check in this block to make sure the complementary boundary of a periodic boundary be periodic (i.e. if the x-min surface is set to periodic, the x-max surface must also be periodic)? We do this in OpenMOC since it doesn't make much sense for only one of the surfaces to be periodic.